src/wallet/types.ts

```diff
@@ -61,6 +61,7 @@ export type UseGetTokenBalanceResponse = {
  */
 export type WalletContextType = {
   address?: Address | null; // The Ethereum address to fetch the avatar and name for.
+  chain?: Chain; // Optional chain for domain resolution
```

Comment:
So, I think we probably want to start move away from using `chain` as type. Mostly because we are seeing occasional missmatch with Viem types.

Let's do instead `chainId` for those things going forward.

Reply:
thoughts on merging this and then updating to chainId in the next PR? Because the chainId change will require `getName`, `Name`, and `useName` updates.

Reply:
I'll push up the chainId changes and we can split it into two PRs if it's too large